evap/staff/tests/test_views.py

```diff
@@ -1914,20 +1914,20 @@ def test_edit_course(self):
         self.course = Course.objects.get(pk=self.course.pk)
         self.assertEqual(self.course.name_en, "A different name")
 
-    @patch("evap.staff.views.redirect")
-    def test_operation_redirects(self, mock_redirect):
-        mock_redirect.side_effect = lambda *_args: HttpResponse()
+    @patch("evap.staff.views.reverse")
+    def test_operation_redirects(self, mock_reverse):
```

Comment:
This now no longer tests that the correct status code is actually returned to the browser. Did you change this because the old mock didn't work anymore? Is it possible that if we simply change [where we patch](https://docs.python.org/3/library/unittest.mock.html#where-to-patch) that we could get stuff to work agian? Otherwise, could we maybe mock the constructor of `HttpResponse` too or something?

Idk, may also not be worth it.

Edit: I see that we simply specify where to redirect to in `def get_success_url` but don't actually do anything about the redirect ourselves. I we argue that we don't want to test django logic, I'd say we should remove the patching completely and simply call `get_success_url` here in the test...

Reply:
Hm, I feel that using `get_success_url` would be too specific to the implementation, best case I would like to get the response and look at it, but by then the urls are resolved already, so I mocked `reverse`. So to include your suggestion, I would say we could either a) keep the current check and also assert that the response redirects (anywhere) or b) only look at the response and compare with the raw url (which we know) or compare with the result of a `reverse(...)`

Reply:
What do you think about the current version?